### PR TITLE
Renaming components: test-procedure to scenario

### PR DESCRIPTION
**Pull request by beaioun**
### Description
Renaming all occurrences of the name test-procedure and test_procedure to scenario, class name TestProcedure to Scenario

### Issues Resolved
issue #308
same as #421 

### Testing
- [x] New functionality includes testing

The changes were tested against an OpenSerach cluster version 1.3.2 for the already merged changes to OS-1 branch on openserach-benchmark-workloads [PR149](https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/149)

> 2024-01-12 01:19:41,839 ActorAddr-(T|:46687)/PID:15931 osbenchmark.test_execution_orchestrator INFO Automatically derived distribution version [1.3.2]
2024-01-12 01:19:42,155 ActorAddr-(T|:46687)/PID:15931 osbenchmark.utils.repo INFO Checking out [OS-1] in [/home/will/.benchmark/benchmarks/workloads/default] for distribution version [1.3.2].
2024-01-12 01:19:42,192 ActorAddr-(T|:46687)/PID:15931 osbenchmark.utils.process INFO Switched to a new branch 'OS-1'
Branch 'OS-1' set up to track remote branch 'OS-1' from 'origin'.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by IanHoang** [comment]
Note: Tests are failing due to workloads repository mainline not having changes. If these changes are added to mainline, they would be introducing breaking changes. 

As an alternative, exploring using github actions in forked repository to test the changes

**Comment by beaioun** [comment]
The changes were tested in this [github action](https://github.com/beaioun/opensearch-benchmark/actions/runs/7632618102) using my forked [repository](https://github.com/beaioun/opensearch-benchmark-workloads) as default workload repository URL. I had cherry picked the changes from OS-1 ([PR #149](https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/149)) to OS-2 and main to match the two versions: `1.3.9, 2.5.0` that this action is running on. The only error now is that the workload-revision points to a past commit in the list_test.py but it shouldn't affect anything. The changes are ready to be populated in opensearch-benchmark-workloads and then this PR should be good to go.

File: MAINTAINERS.md
Comment (cgchinmay) [inline MAINTAINERS.md:1]: Could you check if you have rebased from latest changes. Not sure why this file is included as part of this change

**Comment by IanHoang** [comment]
Closing this PR for now. We will readdress this when gearing up for 2.0.0 release

**Pull request closed without merging**